src/Failed/RabbitQueueFailedJobProvider.php

```diff
@@ -120,7 +120,7 @@ public static function generateMessageId(...$arguments): int
     public function find($id)
     {
         foreach ($this->all() as $job) {
-            if (\property_exists($job, 'id') && $job->id === $id) {
+            if (\property_exists($job, 'id') && \is_numeric($id) && ((int) $job->id) === ((int) $id)) {
```

Comment:
I'm not sure that method's parameter `$id` always should be numeric (`123` or `"123"`). But message's id generation method seems to return integer values. Maybe no need to check `\is_numeric(...)`. Need to discuss.